src/index.js

```diff
@@ -1,12 +1,83 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import * as serviceWorker from './serviceWorker';
-
-ReactDOM.render(<App />, document.getElementById('root'));
-
-// If you want your app to work offline and load faster, you can change
-// unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
-serviceWorker.unregister();
+import React from "react";
+import ReactDOM from "react-dom";
+import { createStore } from "redux";
+import { composeWithDevTools } from "redux-devtools-extension";
+import { Provider } from "react-redux";
+
+import "./styles/main.scss";
+
+import App from "./App";
+const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, ""];
+const winCombo = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, ""];
+
+function shuffleArr(arr) {
+  let newArr = arr;
+  for (let i = arr.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    let temp = arr[i];
+    arr[i] = arr[j];
+    arr[j] = temp;
+  }
+
+  return newArr;
+}
+
+const initialItems = shuffleArr(array);
+
+const setArrayImmutable = (items, index = null, value, empty) => {
+  let newArr = [];
+  if (index === empty + 4) {
+    newArr = Object.assign([...items], { [empty + 4]: "" }, { [empty]: value });
```

Comment:
This type of assignment is not readable. 
- it’s not obvious that `Object.assign` works for arrays too
- it’s not obvious that the result of this call is an array

Suggestion: redo this part with more readable code. Create helper function `immutableSwap(array, clearIndex, fillIndex)`
```
immutableSwap(array, clearIndex, fillIndex) {
  const result = [...array];
  const tempValue = result[clearIndex];
  result[clearIndex] = "";
  result[fillIndex] = tempValue;
  return result;
} 
```

Reply:
Not always one-line-solutions are elegant. The code should be readable and self-explained as well.
I respect your solution but it is not suitable.